### PR TITLE
Split out Angular directives

### PR DESCRIPTION
**Pull request by gisikw**
Started splitting out some of our existing code, based on style guide discussions.

Also:
- Added some directive specs.
- Updated the `from-now` directive to step down from minute-ly polling to hourly polling when it moves to an hour old.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -53,7 +53,16 @@ module.exports = function(config) {
 
       "website/public/js/filters/filters.js",
 
-      "website/public/js/directives/directives.js",
```
Comment (crookedneighbor) [inline karma.conf.js:56]: Is there a reason to not do `"website/public/js/directives/*.js"` or `"website/public/js/directives/**/*.js"`? Other than the obvious, that we're not doing this anywhere else other than in the spec files.

Comment (gisikw) [inline karma.conf.js:56]: Yeah, was just trying to keep it consistent with the manifest.json

File: test/spec/directives/focus-me.directive.spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    scope.$digest();
+  }));
+
+  it('focuses the element when appended to the DOM', function(done) {
+    inject(function($timeout) {
+      element.appendTo(document.body);
+
+      element.on('focus', function() { done() });
+
+      $timeout.flush();
```
Comment (crookedneighbor) [inline test/spec/directives/focus-me.directive.spec.js:24]: This is all very concise, but I think it'd be more human-readable to do something like this:

``` javascript
var focusSpy = sinon.spy();

element.appendTo(document.body);
element.on('focus', focusSpy);
$timeout.flush();

expect(focusSpy).to.be.called;
done();
```

Thoughts?

File: test/spec/directives/from-now.directive.spec.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    });
+
+    element = "<p from-now></p>";
+
+    element = $compile(element)(scope);
+    scope.$digest();
+  }));
+
+  afterEach(function() {
+    window.moment.restore();
```
Comment (crookedneighbor) [inline test/spec/directives/from-now.directive.spec.js:26]: Eventually I'll set up `sinon.sandbox` for these tests :disappointed_relieved:

File: website/public/js/directives/directives.js [hunk 4]
```diff
@@ -1,197 +0,0 @@
...
-   link: function(scope, element, attrs) {
-     var model = $parse(attrs.focusMe);
-     scope.$watch(model, function(value) {
-       $timeout(function() {
-         element[0].focus();
-       });
-     });
-   }
- };
-}]);
```
Comment (crookedneighbor) [inline website/public/js/directives/directives.js:197]: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: 

(about time)

File: test/spec/directives/focus-me.directive.spec.js (hunk 2 above)
Comment (gisikw) [inline test/spec/directives/focus-me.directive.spec.js:24]: My only fear is that relies on ngMock making async behavior "synchronous", which leaves it vulnerable should anything change. Happy to refactor the spec, but if we're testing an async method, it really should be with an async test. Thoughts?

File: karma.conf.js (hunk 1 above)
Comment (crookedneighbor) [inline karma.conf.js:56]: That's a good point. I wonder if there's a way in the manifest.json to easily do `.../*.js` syntax?

File: test/spec/directives/focus-me.directive.spec.js (hunk 2 above)
Comment (crookedneighbor) [inline test/spec/directives/focus-me.directive.spec.js:24]: I just feel weird having a spec without any assertions. You know?

It makes sense to me, but it still feels weird.

Comment (gisikw) [inline test/spec/directives/focus-me.directive.spec.js:24]: Fair enough. I looked briefly at using some of the chai-as-promised functionality, but that didn't play out. Fixed in 9cc4993

**Pull request merged**